File: BAEKJOON/data_structure/javascript/10828.js
const fs = require("fs");
const localPath = __dirname + "/input.txt"; // 로컬 실행용 경로
const baekPath = "/dev/stdin"; // 백준 제출용 경로
let input = fs.readFileSync(localPath).toString().trim().split("\n");
input.shift();
input = input.map((value) => value.split(" "));
const result = [];

class Stack {
  constructor() {
    this.current = [];
  }

  push(value) {
    return this.current.push(value);
  }

  pop() {
    if (this.empty() === 1) {
      return -1;
    }

    return this.current.pop();
  }

  size() {
    return this.current.length;
  }

  empty() {
    if (this.current.length === 0) {
      return 1;
    }

    return 0;
  }

  top() {
    const currentLength = this.current.length;

    if (currentLength === 0) {
      return -1;
    }

    return this.current[currentLength - 1];
  }
}

const testStack = new Stack();

for (let command of input) {
  switch (command[0]) {
    case "push":
      testStack.push(command[1]);
      break;

    case "pop":
      result.push(testStack.pop());
      break;

    case "size":
      result.push(testStack.size());
      break;

    case "empty":
      result.push(testStack.empty());
      break;

    case "top":
      result.push(testStack.top());
      break;
  }
}

console.log(result.join("\n"));

File: BAEKJOON/data_structure/python/10828.py
""" 
22.07.05
BOJ : 10828 / 자료구조
"""
import sys

class my_stack:
    def __init__(self):
        self.result = []
    
    def push(self, num):
        self.result.append(num)
        return 0
    
    def my_pop(self):
        if not self.result:
            print(-1)
            return -1
        else:
            temp = self.result.pop()
            print(temp)
            return 0
    
    def size(self):
        print(len(self.result))
        return 0
    
    def empty(self):
        if not self.result:
            print(1)
            return 0
        else:
            print(0)
            return 0
    
    def top(self):
        if not self.result:
            print(-1)
            return -1
        else:
            print(self.result[-1])
            return self.result[-1]

n = int(input())
answer = my_stack()

for _ in range(n):
    commands = sys.stdin.readline().strip()
    if 'push' in commands:
        answer.push(int(commands.split(' ')[-1]))
    elif 'pop' in commands:
        answer.my_pop()
    elif 'size' in commands:
        answer.size()
    elif 'empty' in commands:
        answer.empty()
    elif 'top' in commands:
        answer.top()
